src/univers/version_range.py

```diff
@@ -1223,14 +1271,142 @@ def build_range_from_github_advisory_constraint(scheme: str, string: str):
+    For example::
+
+    >>> assert split_req_bracket_notation(" 2.3 ]") == ("<=", "2.3")
+    >>> assert split_req_bracket_notation("( 3.9") == (">", "3.9")
+    """
+    comparators_front = {"(": ">", "[": ">="}
+    comparators_rear = {")": "<", "]": "<="}
+
+    constraint_string = remove_spaces(string).strip()
+
+    for native_comparator, vers_comparator in comparators_front.items():
+        if constraint_string.startswith(native_comparator):
+            version = constraint_string.lstrip(native_comparator)
+            return vers_comparator, version
+
+    for native_comparator, vers_comparator in comparators_rear.items():
+        if constraint_string.endswith(native_comparator):
+            version = constraint_string.rstrip(native_comparator)
+            return vers_comparator, version
+
+    raise ValueError(f"Unknown comparator in version requirement: {string!r} ")
+
+
+def build_range_from_snyk_advisory_string(scheme: str, string: Union[str, List]):
+    """
+    Return a VersionRange built from a ``string`` single or multiple snyk
+    version relationship string.
+    Snyk version range looks like:
+        ">=4.0.0, <4.0.10.16"
+        ">=4.1.0 <4.4.15.7"
+        "[3.0.0,3.1.25)"
+        "(,9.21]"
+        "[1.4.5,)"
+
+    For example::
+
+    >>> vr = build_range_from_snyk_advisory_string("pypi", ">=4.0.0, <4.0.10")
+    >>> assert str(vr) == "vers:pypi/>=4.0.0|<4.0.10"
+    >>> vr = build_range_from_snyk_advisory_string("golang", ">=9.6.0-rc1 <9.8.1-rc1")
+    >>> assert str(vr) == "vers:golang/>=9.6.0-rc1|<9.8.1-rc1"
+    >>> vr = build_range_from_snyk_advisory_string("pypi", "(,9.21]")
+    >>> assert str(vr) == "vers:pypi/<=9.21"
+    """
+    version_constraints = []
+    vrc = RANGE_CLASS_BY_SCHEMES[scheme]
+
+    if isinstance(string, str):
+        string = [string]
+
+    for item in string:
+        delimiter = "," if "," in item else " "
+        if delimiter == ",":
+            snyk_constraints = item.strip().replace(" ", "")
+            constraints = snyk_constraints.split(",")
+        else:
+            snyk_constraints = item.strip()
+            constraints = snyk_constraints.split(" ")
+
+        for constraint in constraints:
+            if any(comp in constraint for comp in "[]()"):
+                comparator, version = split_req_bracket_notation(string=constraint)
+            else:
+                comparator, version = split_req(
+                    string=constraint,
+                    comparators=vers_by_snyk_native_comparators,
+                )
+            if comparator and version:
+                version = vrc.version_class(version)
+                version_constraints.append(
+                    VersionConstraint(
+                        comparator=comparator,
+                        version=version,
+                    )
+                )
+    return vrc(constraints=version_constraints)
+
+
+def build_range_from_discrete_version_string(scheme: str, string: Union[str, List]):
```

Comment:
Would this be a possible dupe of https://github.com/nexB/univers/blob/8f0d7276e13686d93fee5313641b9803f4d4cb15/src/univers/version_range.py#L166 ?

Reply:
Yup, droping it in favour of from_versions.